File: PeopleSearch/Models/Person.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PeopleSearch.Models
{
    /// <summary>
    /// Represents a person.
    /// </summary>
    public class Person
    {
        /// <summary>
        /// A database-generated ID.
        /// </summary>
        public long Id { get; set; }

        /// <summary>
        /// The person's first name. This field is required and limited to 50 characters.
        /// </summary>
        [Required]
        [Display(Name = "First Name")]
        [StringLength(50)]
        public string FirstName { get; set; }

        /// <summary>
        /// The person's last name. This field is required and limited to 50 characters.
        /// </summary>
        [Required]
        [Display(Name = "Last Name")]
        [StringLength(50)]
        public string LastName { get; set; }

        /// <summary>
        /// The person's Address.
        /// </summary>
        public Address Address { get; set; }

        /// <summary>
        /// The person's birth date. The time is not persisted. This field is required.
        /// </summary>
        [Required]
        [Display(Name = "Birth Date")]
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:MM-dd-yyy}", ApplyFormatInEditMode = true)]
        public DateTime BirthDate { get; set; }

        /// <summary>
        /// The person's death date, if applicable. The time is not persisted. This field in optional.
        /// </summary>
        [Display(Name = "Death Date")]
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:MM-dd-yyy}", ApplyFormatInEditMode = true)]
        public DateTime? DeathDate { get; set; }

        /// <summary>
        /// The person's age derived from either the current date or their death date.
        /// </summary>
        [NotMapped]
        public TimeSpan Age => (DeathDate ?? DateTime.Now) - BirthDate;

        /// <summary>
        /// A list of the person's interests.
        /// </summary>
        public IList<Interest> Interests { get; set; }

        /// <summary>
        /// The person's image.
        /// </summary>
        public Image Image { get; set; }
    }
}

File: PeopleSearch/src/app/models/person.ts
import { Address } from './address';
import { Interest } from './interest';
import { Image } from './image';

export class Person {
  constructor(data: any = {}) {
    Object.assign(this, data);
  }

  id: number = 0;
  image: Image;
  firstName: string = '';
  lastName: string = '';
  birthDate: Date = new Date(Date.now());
  deathDate: Date;
  address: Address = new Address();
  interests: Interest[] = [];

  get age(): number {
    var endDate = this.deathDateAsDate() || new Date(Date.now());
    var startDate = this.birthDateAsDate();
    var ageDifMs = endDate.getTime() - startDate.getTime();
    var ageDate = new Date(ageDifMs); // miliseconds from epoch
    return Math.abs(ageDate.getUTCFullYear() - 1970);
  }

  birthDateAsDate(): Date {
    return new Date(this.birthDate);
  }

  deathDateAsDate(): Date {
    return this.deathDate ? new Date(this.deathDate) : null;
  }
}
